mlir/include/mlir/Dialect/Linalg/IR/LinalgRelayoutOps.td

```diff
@@ -76,6 +76,13 @@ class Linalg_RelayoutOp<string mnemonic, list<Trait> traits = []> :
     /// have been tiled. Also, the order of the output dimensions is consistent
     /// with `inner_dims_pos` rather than the packed tensor.
     SmallVector<int64_t> getTiledOuterDims();
+
+         void $cppClass::getEffects(
+         SmallVectorImpl<SideEffects::EffectInstance<MemoryEffects::Effect>>
+             &effects) {
+       getGenericEffectsImpl(effects, cast<LinalgOp>(getOperation()));
+     }
```

Comment:
I think it is better to put the implementation to LinalgOps.cpp. And I believe that this is not working for relayout ops because they are not LinalgOp. I think you need to implement it separately and you can follow what we've done in IREE.

https://github.com/llvm/llvm-project/blob/6075275e68bd2362095af76b1acb49c2ac9610a4/mlir/include/mlir/Dialect/Linalg/IR/LinalgInterfaces.td#L225-L229

Reply:
I implemented getEffects, but their implementations are identical for packOp, unpackOp, I think it would be great to wrap somewhere, could you suggest some?

Reply:
Can we do it with template? E.g., they have the identical implementation for `reifyResultShapes`, and we have the implementation in `reifyResultShapesImpl`. I think we can use the same trick here.

https://github.com/llvm/llvm-project/blob/8244f8210f2e62f68429a0daf104fd483ada45ab/mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp#L4330-L4341

Reply:
Thanks! Done!